### PR TITLE
Add distribution and request csvs to commandline usage.

### PR DESCRIPTION
**Pull request by cpennington**
Original-Author: Kevin Peter Savage (https://github.com/kevinpetersavage)

### CODE REVIEW

**Comment by cpennington** [comment]
@cgoldberg @alawibaba @BenjiLee @jimabramson: Review?

File: locust/stats.py [hunk 1]
```diff
@@ -392,6 +392,67 @@ def from_dict(cls, data):
...
+            '"100%"',
+        ))]
+        for s in chain(
+                (stats for (name, stats) in sorted(cls.requests.iteritems())),
+                [cls.sum_stats("Total", full_request_history=True)]
+            ):
+            if s.num_reqs:
+                rows.append(s.percentile(tpl='"%s",%i,%i,%i,%i,%i,%i,%i,%i,%i,%i'))
+            else:
+                rows.append('"%s",0,"N/A","N/A","N/A","N/A","N/A","N/A","N/A","N/A","N/A"' % s.name)
```
Comment (cgoldberg) [inline locust/stats.py]: nit:  use format() instead of old-school string formatting

File: locust/stats.py [hunk 2]
```diff
@@ -392,6 +392,67 @@ def from_dict(cls, data):
...
+                '"Average Content Size"',
+                '"Reqests/s"',
+            ])
+        ]
+
+        for s in chain(
+                (stats for (name, stats) in sorted(cls.requests.iteritems())),
+                [cls.sum_stats("Total", full_request_history=True)]
+            ):
+            rows.append('"%s","%s",%i,%i,%i,%i,%i,%i,%i,%.2f' % (
```
Comment (cgoldberg) [inline locust/stats.py]: nit: use format() instead of old-school string formatting

**Comment by cgoldberg** [comment]
lgtm

**Comment by BenjiLee** [comment]
Looks good so far. Going to try it before I thumbs up.

File: locust/main.py [hunk 3]
```diff
@@ -228,6 +228,15 @@ def parse_options():
         help="show program's version number and exit"
     )
 
+    # Writes output statistics as csv
+    parser.add_option(
+        '--write-stats',
+        action='store_true',
+        dest='write_stats',
+        default=False,
+        help="writes output statistics as csv files"
```
Comment (cpennington) [inline locust/main.py]: This should take an optional path prefix to write the files to.

File: locust/main.py [hunk 4]
```diff
@@ -290,6 +299,19 @@ def is_locust(tup):
     )
 
 
+def write_distribution_stats_csv():
+    stamp = time()
+    file_name = "distribution_{0}.csv".format(stamp)
+    with open(file_name, 'w') as file:
```
Comment (cpennington) [inline locust/main.py]: This should log a message about where it's writing to.

File: locust/runners.py [hunk 5]
```diff
@@ -5,6 +5,7 @@
 import random
 import logging
 from time import time
+from itertools import chain
```
Comment (jimabramson) [inline locust/runners.py]: shouldn't this import be in stats.py ?

**Comment by jimabramson** [comment]
i think it's weird not to be using the csv library for this, but as far as unblocking the command-line use case with a minimum of change, lgtm.

**Comment by cpennington** [comment]
I actually set it up to use the csv library, and then converted back when I couldn't figure out how to make flask give me a file-handle for the response. The csv library is very focused on file-based output.

**Comment by cpennington** [comment]
Now depends on #4.

**Comment by jimabramson** [comment]
for some reason i can't find my previous comments on this PR but looks like everything got addressed...
:+1:

**Comment by cpennington** [comment]
@jimabramson you commented on a particular commit, and then I rebased.

**Pull request merged**